mp/src/game/server/momentum/mom_triggers.cpp

```diff
@@ -1497,10 +1502,14 @@ bool CNoGrenadesZone::IsInsideNoGrenadesZone(CBaseEntity *pOther)
 {
     if ( pOther )
     {
+        CMomStickybomb* pSticky = dynamic_cast<CMomStickybomb*>(pOther);
         FOR_EACH_VEC(s_vecNoGrenadeZones, i)
         {
             const auto pNoGrenadeZone = s_vecNoGrenadeZones[i];
-            if (!pNoGrenadeZone->m_bDisabled && pNoGrenadeZone->PointIsWithin(pOther->GetAbsOrigin()))
+
+            bool bLandedStickyInWallpogoNonades = pSticky ? (pSticky->GetUseImpactNormal() && pNoGrenadeZone->m_bAirborneOnly) : false;
+
+            if (!pNoGrenadeZone->m_bDisabled && pNoGrenadeZone->PointIsWithin(pOther->GetAbsOrigin()) && !bLandedStickyInWallpogoNonades)
                 return true;
```

Comment:
I think this can be cleaned up a bit more.
```suggestion
            if (pNoGrenadeZone->m_bDisabled || !pNoGrenadeZone->PointIsWithin(pOther->GetAbsOrigin())
                continue;
            
            if (!pSticky)
                return true; // Rocket exploding in nogrenade zone
            
            return !pNoGrenadeZone->m_bAirborneOnly || !pSticky->DidHitWorld();
```